modules/ROOT/pages/dataweave-cookbook-decode-encode.adoc

```diff
@@ -0,0 +1,160 @@
+----
+%PDF-1.6
+%����
+37 0 obj <</Linearized 1/L 20597/O 40/E 14115/N 1/T 19795/H [ 1005 215]>>
+endobj
+                 
+xref
+37 34
+0000000016 00000 n
+...
+----
+
+[[xml_config]]
+== XML Configuration Example
+
+In Anypoint Studio, you can create and run a Mule flow that encodes and decodes a PDF file. After you complete the procedure, the example will read the file from the specified local directory using the Read operation from the File module. It will write the PDF to the specified directory by using the Write operation in the File module. It will use Transform Message components to encode the file to Base64 and to decode that output from Base64.
+
+. Create a Mule project in Anypoint Studio.
+. Use *Add Module* in the Mule Palette to add the File module.
+. Copy and paste the following example between the `<mule/>` elements of a Mule flow in Anypoint Studio.
++
+[source,XML,linenums]
+----
+<http:listener-config name="HTTP_Listener_config"
+                      doc:name="HTTP Listener config" >
+    <http:listener-connection
+                            host="0.0.0.0"
+                            port="8081" />
+</http:listener-config>
+<file:config name="File_Config"
+             doc:name="File Config" >
+    <file:connection workingDir="/Users/me/Downloads" />
+</file:config>
+<flow name="encode-decode-base64">
+    <http:listener doc:name="Listener"
+                 path="/transform"
+                 config-ref="HTTP_Listener_config"/>
+    <file:read doc:name="Read"
+             config-ref="File_Config"
+             path="pdf-test.pdf" />
+    <ee:transform doc:name="Transform Message">
+        <ee:message>
+          <ee:set-payload><![CDATA[%dw 2.0
+import * from dw::core::Binaries
+output text/plain
+---
+toBase64(payload as Binary)]]></ee:set-payload>
+        </ee:message>
+    </ee:transform>
+    <ee:transform doc:name="Transform Message" >
+        <ee:message >
+          <ee:set-payload ><![CDATA[%dw 2.0
+import * from dw::core::Binaries
+output application/pdf with binary
+---
+fromBase64(payload as String) as Binary]]></ee:set-payload>
+        </ee:message>
+    </ee:transform>
+    <file:write doc:name="Write"
+              path="/Users/me/pdf-test-result.pdf"/>
+    <set-payload
+              value='#["PDF file created in the directory specified in the Write operation."]'
+              doc:name="Set Payload" />
+</flow>
+----
++
+Note that it is valid to replace the directive `output application/pdf with binary` in the second Transform Message component with `output application/octet-stream`.
++
+. Place a PDF file of your choice into a local directory on your machine.
+. Correct the input and output directories of the Read and Write operations:
++
+* In the Read operation, change the *File Path* value `/Users/me/Downloads" to the location of your PDF file.
+* In the Write operation, change the *Path* value `/Users/me/pdf-test.pdf"` to the location and file name that you want to output the file.
```

Comment:
```suggestion
* In the *Read* operation, change the *File Path* value `/Users/me/Downloads" to the location of your PDF file.
* In the *Write* operation, change the *Path* value `/Users/me/pdf-test.pdf"` to the location and file name that you want to output the file.
```

Reply:
In Connectors we style the operation name in bold, just adding it here for consistency.

Reply:
done, thx for heads up